components-core/src/main/java/com/adyen/checkout/components/core/extension/PaymentMethodsApiResponseExtension.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ararat on 6/11/2023.
+ */
+
+package com.adyen.checkout.components.core.extension
+
+import com.adyen.checkout.components.core.PaymentMethod
+import com.adyen.checkout.components.core.PaymentMethodCustomDisplayInformation
+import com.adyen.checkout.components.core.PaymentMethodsApiResponse
+import com.adyen.checkout.components.core.StoredPaymentMethod
+
+/**
+ * Allows setting custom display information for payment methods, allowing filter by [type] and [predicate].
+ *
+ * Calling this function multiple times will cause the custom display information to be overridden.
+ * This might be useful when [customDisplayInformation] should be updated when localization settings change.
+ *
+ * @param type Updates payment methods matching the given type.
+ * @param customDisplayInformation Customizable information object to override the default display values.
+ * @param predicate Updates payment methods not only matching the type but also matching the given predicate.
+ */
+fun PaymentMethodsApiResponse.addPaymentMethodCustomDisplayInformation(
+    type: String,
```

Comment:
Wouldn't it be easier if the merchant just passes the `PaymentMethod`? It would be less flexible, but also less error prone 🤔

Reply:
It is possible indeed, but in that case they need to filter the payment methods themselves, which might add some extra code when using the `addPaymentMethodCustomDisplayInformation` method. In that case we might also need to support passing payment methods list to the `addPaymentMethodCustomDisplayInformation` method, if their filter result is a list.
Perhaps what we could do is modify the `PaymentMethodTypes` object, by making it an `enum` (by also allowing custom types) and asking merchants to pass an enum value instead of a String for `type`?

Reply:
Ah yes didn't take about the filtering yet. An enum would probably be best indeed as it covers both sides. Let's discuss if and how we could do that